### DISCUSSION TITLE
Team Meeting 27/11/22

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#8472] [General]
## Topics
- [ ] mixins and classes is one i care most, also necessary to finish this part of the migration, and other PRs
   - [x] migrate canvas + staticCanvas
   - [x] filters #8474 
   - [x] #8414 ( review + merge )
   - [x] #8461 
   - [x] #8297  ( verify it doesn't break stateful animation )
   - [ ]  #8374 ( needs specific review )
   - [x] #8282  ( Andrea to counter propose )
   - [ ] etc
- [ ] the migration path, meaning what you have in mind next, when/how we finish, steps, what is in and what is out etc.
   - [ ] jsdocs -> look in support for TS and dedupe documentation efforts from TS
   - [ ] transpiling
   - [ ] supported browsers ( with an eye to transpiling we are going to use a generic supported browser statement major by major )
   - [ ] rc publish flow and versioning ( investigate how to git tag versions in the repo ( and package.json )) #8479 
- [x] Contexts (browser, node, SSR, etc.) and the import pattern ( discussed, some investigation to do )
- [ ] [ci](https://github.com/fabricjs/fabric.js/pulls?q=is%3Aopen+is%3Apr+label%3ACI%2FCD)
  - [ ] [github token for forked PRs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#granting-additional-permissions)
  - [ ] [codesandbox CI](https://codesandbox.io/docs/learn/integrations/github-app)
  - [ ] [Gitpod App](https://probot.github.io/apps/gitpod-io/)
- [ ] Brush API
- [ ] draft a long term topic list

**Comment by asturur** [comment]
Some topics we touched were:
- investigate how to move bounding boxes calculations in specific workers and remove the getTransformedDimensions concept that isn't flexibile anymore
- same for group layouts

**Comment by ShaMan123** [comment]
Versioning:
- https://github.com/marketplace/actions/tag-release-on-push-action: doesn't change package.json but accepts PR labels for versioning
- https://github.com/marketplace/actions/github-tag: I think it does change package.json but doesn't accept PR labels for versioning
- https://github.com/marvinpinto/action-automatic-releases - only prereleases

All 3 seem insufficient so we can impl ourselves or patch one of them
I think no. 2 is the best option...
Thoughts?

**Reply by ShaMan123** [reply]
Actually I am not sure, maybe the first is best.
Do we have to maintain the version in package.json? Cause we can obtain the tag it in the release action... can that be done with npm? If it needs package.json version we can modify it in the release action and that's solved.
So in light of the above I am for the first action.

**Reply by ShaMan123** [reply]
#8479
